File: test-fixtures/languages/ruby/micro/user_service.rb
require_relative 'auth_service'

module UserService
  def self.get_user(token)
    user_id = AuthService.verify_token(token)
    { id: user_id, token: token }
  end

  def self.update_user(token, action)
    user_id = AuthService.verify_token(token)
    if user_id
      AuthService.run_admin_command(user_id, action)
    end
    user_id
  end
end

File: test-fixtures/languages/scala/micro/UserService.scala
package micro

case class UserInfo(id: Option[String], token: String)

class UserService {
  private val authService = new AuthService()

  def getUser(token: String): UserInfo = {
    val userId = authService.verifyToken(token)
    UserInfo(id = userId, token = token)
  }

  def updateUser(token: String, action: String): Option[String] = {
    val userId = authService.verifyToken(token)
    val uid = userId.getOrElse("")
    authService.runAdminCommand(uid, action)
    userId
  }
}
